evalml/pipelines/components/estimators/regressors/arima_regressor.py

```diff
@@ -140,14 +160,20 @@ def fit(self, X, y=None):
             self
 
         Raises:
-            ValueError: If X was passed to `fit` but not passed in `predict`.
+            ValueError: If y was not passed in.
         """
         if X is not None:
             X = X.fillna(X.mean())
         X, y = self._manage_woodwork(X, y)
         if y is None:
             raise ValueError("ARIMA Regressor requires y as input.")
 
+        parameters = self.arima_parameters
+        parameters["sp"] = self._get_sp(X)
+
+        arima_model = self.sktime_arima.AutoARIMA(**parameters)
+        self._component_obj = arima_model
```

Comment:
So, is the reason we deviated from letting the parent class set the `_component_obj` was because we needed to see the X data in fit and use the frequency of the data to set `sp`?  

I'm a little shaky on that approach.  After we instantiate an `ARIMARegressor`, it, unlike other regressors/estimators, will have no `_component_obj`.  Perhaps we could try keeping the passing of the model, with most of the parameters set in `__init__()` to the parent's constructor to set the `_component_obj` but then call `set_params()` [docs](https://alkaline-ml.com/pmdarima/modules/generated/pmdarima.arima.AutoARIMA.html#pmdarima.arima.AutoARIMA.set_params) after using `.clone()` or `.get_params` on the AutoARIMA forecaster in `.fit()`

Reply:
Gotcha - I was a little shaky as well, but I was worried setting a parameter after instantiation would affect the results. I've switched it back to instantiating the `component_obj` in init and everything still seems to work fine, so I'll leave it there.